### PR TITLE
[RHPAM-2310] Add process migration module

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

Fix https://issues.jboss.org/browse/RHPAM-2310

Add cekit module for process migration service

### CODE REVIEW

**Comment by jakubschwan** [comment]
@ruromero please add label "on hold" or "wip". RHPAM-2310 targeting 7.6.x release and now we still not have 7.5.x branch for this repo. We need to wait now with the merge of this PR.

File: process-migration/modules/process-migration/module.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+schema_version: 1
+name: "rhpam-7-process-migration"
+description: "Red Hat Process Automation Manager Process Migration 7.5 install"
+labels:
+- name: "org.jboss.product"
+  value: "rhpam-process-migration"
+- name: "org.jboss.product.version"
+  value: "7.5.0"
```
Comment (errantepiphany) [inline process-migration/modules/process-migration/module.yaml]: change to "7.6.0" eveywhere

File: process-migration/modules/process-migration/module.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- name: "JBOSS_PRODUCT"
+  value: "rhpam-process-migration"
+- name: "RHPAM_PROCESS_MIGRATION_VERSION"
+  value: "7.5.0"
+- name: "PRODUCT_VERSION"
+  value: "7.5.0"
+- name: "ADD_ONS_DISTRIBUTION_ZIP"
+  value: "add_ons_distribution.zip"
+- name: "KIE_PROCESS_MIGRATION_DISTRIBUTION_JAR"
+  value: "rhpam-7.5-process-migration-service-standalone.jar"
```
Comment (errantepiphany) [inline process-migration/modules/process-migration/module.yaml]: change to "7.6" eveywhere

File: process-migration/tag-overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#          at reliable and unchanging reference points. Useful for CVE respins.
+#          Warning: Only reliable if referenced via a tag itself!
+#   Usage: cekit build --overrides-file tag-overrides.yaml
+
+modules:
+      repositories:
+          - name: cct_module
+            git:
+                  url: https://github.com/jboss-openshift/cct_module.git
+                  ref: sprint-35
```
Comment (errantepiphany) [inline process-migration/tag-overrides.yaml]: change to "0.35.1"

File: process-migration/image.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+schema_version: 1
+
+name: "rhpam-7/rhpam75-process-migration"
+description: "Red Hat Process Automation Manager Process Migration 7.5 container image"
+version: "7.5.0"
+from: "ubi8:8-released"
```
Comment (spolti) [inline process-migration/image.yaml]: maybe use ubi minimal?

File: process-migration/image.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+labels:
+    - name: "com.redhat.component"
+      value: "rhpam-7-rhpam75-process-migration-rhel8-container"
```
Comment (spolti) [inline process-migration/image.yaml]: isn't better just rhpam-process-migration-container?

File: process-migration/image.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+            version: "11"
+          - name: rhpam-7-process-migration
+packages:
+      content_sets_file: content_sets.yml
+osbs:
+      configuration:
+          container_file: container.yaml
+      repository:
+          name: containers/rhpam-7-process-migration
+          branch: rhba-7.5-rhel-8
```
Comment (spolti) [inline process-migration/image.yaml]: 7.6

File: process-migration/tests/features/process-migration.feature [hunk 7]
```diff
@@ -0,0 +1,10 @@
+@rhpam-7/rhpam75-process-migration
+Feature: RHPAM Process Migration tests
+
+   Scenario: Test REST API is available and valid
+     When container is ready
+     Then check that page is served
+         | property | value |
+         | port     | 8080  |
+         | path     | /rest/health/status |
+         | expected_phrase | SUCCESS |
```
Comment (spolti) [inline process-migration/tests/features/process-migration.feature]: is this test applicable to this new container?

Also, is there more tests that can be done?

Comment (ruromero) [reply process-migration/tests/features/process-migration.feature]: This one should be enough, it is the health probe that will be used when deployed

File: process-migration/image.yaml (hunk 5 above)
Comment (ruromero) [reply process-migration/image.yaml]: Following the same pattern as the other images. Isn't it correct?

Comment (errantepiphany) [reply process-migration/image.yaml]: For now I say follow the same pattern as the other images We can revisit changing all of them if need be.

**Comment by ruromero** [comment]
@errantepiphany @spolti I have rebased. Please check also the change on the smartrouter modue.yaml

**Comment by errantepiphany** [comment]
@ruromero Can you please resolve the conflicts and squash your commits before we merge this?

File: process-migration/tests/features/process-migration.feature [hunk 8]
```diff
@@ -0,0 +1,10 @@
+@rhpam-7/rhpam75-process-migration
```
Comment (errantepiphany) [inline process-migration/tests/features/process-migration.feature]: Oops I think we missed a "75" to "76" bump here. :)

**Pull request merged**